astropy/io/fits/convenience.py

```diff
@@ -494,6 +494,8 @@ def table_to_hdu(table):
         fits_rec = FITS_rec.from_columns(np.array(table.filled()))
         table_hdu = BinTableHDU(fits_rec)
```

Comment:
Oh it's probably because of this: the table should be created with `BinTableHDU.from_columns` instead of these 2 lines.